### ISSUE TITLE
Unparsable block

### ISSUE THREAD

**Issue by Ferrany1** [sqlfluff/sqlfluff#6488]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In migration:

```sql
CREATE schema test;

CREATE TABLE test.test (
    id UUID PRIMARY KEY,

    created_at TIMESTAMP WITH TIME ZONE DEFAULT (NOW() AT TIME ZONE 'UTC') NOT NULL,
    updated_at TIMESTAMP WITH TIME ZONE DEFAULT (NOW() AT TIME ZONE 'UTC') NOT NULL
)
```
Command:
```
running sqlfluff lint up.sql --dialect postgres
```

Error :
```
Found unparsable section: 'CREATE TABLE test.test (\\n    id...'
```

### Expected Behaviour

Block with (NOW() AT TIME ZONE 'UTC') parsed correctly

### Observed Behaviour

Block (NOW() AT TIME ZONE 'UTC') can't be parsed

### How to reproduce

```up.sql
CREATE schema test;

CREATE TABLE test.test (
    id UUID PRIMARY KEY,

    created_at TIMESTAMP WITH TIME ZONE DEFAULT (NOW() AT TIME ZONE 'UTC') NOT NULL,
    updated_at TIMESTAMP WITH TIME ZONE DEFAULT (NOW() AT TIME ZONE 'UTC') NOT NULL
)
```
Command:
```
running sqlfluff lint up.sql --dialect postgres
```

### Dialect

postgres

### Version

```
sqlfluff --version                                                                                                                                                                            ✭ ✱
sqlfluff, version 3.2.5
```

### Configuration

No extra configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joaostorrer** [comment]
I tested with version 3.3.0 and it's parsing without error.

**Comment by keraion** [comment]
I was unable to reproduce the error from the example in version 3.2.5. Closing out as is.
